src/core/layertree/qgslayertreemodellegendnode.cpp

```diff
@@ -490,6 +514,42 @@ QSizeF QgsSymbolLegendNode::drawSymbol( const QgsLegendSettings &settings, ItemC
                  std::max( height + 2 * heightOffset, static_cast< double >( settings.symbolSize().height() ) ) );
 }
 
+void QgsSymbolLegendNode::drawSymbol( const QgsLegendSettings &settings, QJsonObject &json ) const
```

Comment:
This should really accept an explicit QgsRenderContext argument - otherwise symbols won't be rendered correctly (e.g. not respecting the correct expression context), and things like the new support for expression based symbol text and upcoming change to QgsTextRenderer to draw legend text won't work correctly.

Reply:
I'm glad you spotted that one Nyall!

Reply:
Done